core/templates/dev/head/collection_editor/editor_tab/CollectionLinearizerServiceSpec.js

```diff
@@ -0,0 +1,468 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Tests for CollectionLinearizerService.
+ */
+
+describe('Collection linearizer service', function() {
+  var CollectionObjectFactory = null;
+  var CollectionNodeObjectFactory = null;
+  var CollectionLinearizerService = null;
+
+  var firstCollectionNode = null;
+  var secondCollectionNode = null;
+  var thirdCollectionNode = null;
+
+  beforeEach(module('oppia'));
+
+  beforeEach(inject(function($injector) {
+    CollectionObjectFactory = $injector.get('CollectionObjectFactory');
+    CollectionNodeObjectFactory = $injector.get('CollectionNodeObjectFactory');
+    CollectionLinearizerService = $injector.get('CollectionLinearizerService');
+
+    var firstCollectionNodeBackendObject = {
+      exploration_id: 'exp_id0',
+      prerequisite_skills: [],
+      acquired_skills: [],
+      exploration_summary: {
+        title: 'exp title0',
+        category: 'exp category',
+        objective: 'exp objective'
+      }
+    };
+    firstCollectionNode = CollectionNodeObjectFactory.create(
+        firstCollectionNodeBackendObject);
+
+    var secondCollectionNodeBackendObject = {
+      exploration_id: 'exp_id1',
+      prerequisite_skills: [],
+      acquired_skills: [],
+      exploration_summary: {
+        title: 'exp title1',
+        category: 'exp category',
+        objective: 'exp objective'
+      }
+    };
+    secondCollectionNode = CollectionNodeObjectFactory.create(
+        secondCollectionNodeBackendObject);
+
+    var thirdCollectionNodeBackendObject = {
+      exploration_id: 'exp_id2',
+      prerequisite_skills: [],
+      acquired_skills: [],
+      exploration_summary: {
+        title: 'exp title2',
+        category: 'exp category',
+        objective: 'exp objective'
+      }
+    };
+    thirdCollectionNode = CollectionNodeObjectFactory.create(
+        thirdCollectionNodeBackendObject);
+  }));
+
+  var createLinearCollection = function() {
+    var collection = CollectionObjectFactory.createEmptyCollection();
```

Comment:
Consider adding a comment here describing what the order of explorations will be within the collection, e.g.:

```The exploration order is: exp_id0 -> exp_id1 -> exp_id2```